### ISSUE TITLE
Clean checkout M1 docker-compose up -d fails with ModuleNotFoundError: No module named 'Cython'

### ISSUE THREAD

**Issue by emilebosch** [getredash/redash#5805]
### Issue Summary

Clean checkout Apple M1 pro docker-compose up -d fails with ModuleNotFoundError: No module named 'Cython' on master(4186f8303ee28d5cde7ca248c57df9023eba6224) Docker desktop .6.0 (75818)

```
=> ERROR [redash_worker stage-1  8/15] RUN if [ "x$skip_ds_deps" = "x" ] ; then pip install -r requirements_all_ds.txt ; else echo "Skipping pip install -r requirements_all_ds.txt"  4.0s
------
 > [redash_worker stage-1  8/15] RUN if [ "x$skip_ds_deps" = "x" ] ; then pip install -r requirements_all_ds.txt ; else echo "Skipping pip install -r requirements_all_ds.txt" ; fi:
#17 0.555 Collecting google-api-python-client==1.7.11
#17 0.602   Downloading google-api-python-client-1.7.11.tar.gz (142 kB)
#17 1.083 Collecting protobuf==3.17.3
#17 2.004   Downloading protobuf-3.17.3-cp37-cp37m-manylinux2014_aarch64.whl (922 kB)
#17 2.127 Collecting gspread==3.1.0
#17 2.134   Downloading gspread-3.1.0-py3-none-any.whl (16 kB)
#17 2.161 Collecting impyla==0.16.0
#17 2.168   Downloading impyla-0.16.0.tar.gz (235 kB)
#17 2.390 Collecting influxdb==5.2.3
#17 2.396   Downloading influxdb-5.2.3-py2.py3-none-any.whl (73 kB)
#17 2.434 Collecting mysqlclient==1.3.14
#17 2.440   Downloading mysqlclient-1.3.14.tar.gz (91 kB)
#17 2.624 Collecting oauth2client==4.1.3
#17 2.630   Downloading oauth2client-4.1.3-py2.py3-none-any.whl (98 kB)
#17 2.656 Collecting pyhive==0.6.1
#17 2.663   Downloading PyHive-0.6.1.tar.gz (41 kB)
#17 3.278 Collecting pymongo[srv,tls]==3.9.0
#17 3.287   Downloading pymongo-3.9.0.tar.gz (701 kB)
#17 3.574 Collecting vertica-python==0.9.5
#17 3.581   Downloading vertica_python-0.9.5-py2.py3-none-any.whl (137 kB)
#17 3.621 Collecting td-client==1.0.0
#17 3.628   Downloading td_client-1.0.0-py3-none-any.whl (78 kB)
#17 3.705 Collecting pymssql==2.1.4
#17 3.713   Downloading pymssql-2.1.4.tar.gz (691 kB)
#17 3.927     ERROR: Command errored out with exit status 1:
#17 3.927      command: /usr/local/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-yrvn6v20/pymssql/setup.py'"'"'; __file__='"'"'/tmp/pip-install-yrvn6v20/pymssql/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-17xsw5as
#17 3.927          cwd: /tmp/pip-install-yrvn6v20/pymssql/
#17 3.927     Complete output (7 lines):
#17 3.927     /usr/local/lib/python3.7/site-packages/setuptools/dist.py:51: DistDeprecationWarning: Do not call this function
#17 3.927       warnings.warn("Do not call this function", DistDeprecationWarning)
#17 3.927     Traceback (most recent call last):
#17 3.927       File "<string>", line 1, in <module>
#17 3.927       File "/tmp/pip-install-yrvn6v20/pymssql/setup.py", line 88, in <module>
#17 3.927         from Cython.Distutils import build_ext as _build_ext
#17 3.927     ModuleNotFoundError: No module named 'Cython'
#17 3.927     ----------------------------------------
#17 3.928 ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
------
failed to solve: executor failed running [/bin/sh -c if [ "x$skip_ds_deps" = "x" ] ; then pip install -r requirements_all_ds.txt ; else echo "Skipping pip install -r requirements_all_ds.txt" ; fi]: exit code: 1
```

### Steps to Reproduce

1. Git clone
2. Set .env  with only a cookie var
3. docker-compose up -d

Seems a dep is missing, or at least pip hell.

### Technical details:

* Redash Version: 4186f8303ee28d5cde7ca248c57df9023eba6224
* Browser/OS: Safari
* How did you install Redash: docker-compose up -d

**Comment by susodapop** [comment]
We're aware of issues building on Apple Silicon using the `master` branch.  Try building using the `fix-m1-build` branch here: https://github.com/getredash/redash/tree/fix-m1-build

**Comment by ar-ms** [comment]
This issue arises because of the architecture used by default when building the image on the M1 (ARM architecture) which is not supported currently by every libraries. We can change the default behaviour and build for an x86_64 by specifying the architecture using the platform flag.

In our case, it can be done by modifying the line 28 of the Dockerfile, by simply changing the line from
`FROM python:3.7-slim-buster` to `FROM --platform=linux/amd64 python:3.7-slim-buster`.

**Comment by susodapop** [comment]
Yes that's another valid work around 👌

**Comment by ar-ms** [comment]
I run into another issue, the issue is related to watchdog and inotify, the server and the workers were failing, forgot the exact error message but it was similar to this #5804 .

It seems to be a known issue from Docker (https://docs.docker.com/desktop/mac/apple-silicon/) and watchdog (https://github.com/gorakhargosh/watchdog/issues/838).

> However, attempts to run Intel-based containers on Apple silicon machines under emulation can crash as qemu sometimes fails to run the container. In addition, filesystem change notification APIs (inotify) do not work under qemu emulation. Even when the containers do run correctly under emulation, they will be slower and use more memory than the native equivalent.

Running the scheduler and the workers in a none dev mode resolved the issue, the dev versions are using watchmedo to restart the scheduler and workers files are modified, that's a nice feature but unfortunately I had to get rid of it to make it work on the M1.

**docker-compose.yml**
```yaml
  scheduler:
    <<: *redash-service
    command: scheduler
    depends_on:
      - server
    environment:
      <<: *redash-environment
  worker:
    <<: *redash-service
    command: worker
    depends_on:
      - server
    environment:
      <<: *redash-environment
      PYTHONUNBUFFERED: 0
```

**Comment by arijitAD** [comment]
@ar-ms Were you able to run the docker image on M1 successfully?

**Comment by willgladstone** [comment]
I was able to get it to build with the following changes:

In [requirements_all_ds.txt](https://github.com/getredash/redash/blob/master/requirements_all_ds.txt) 

- Upgrade pymssql: `pymssql==2.1.5`. 
- Remove ibm db2 connectivity (their recommended fix is to run python as amd64): `#ibm-db>=2.0.9 #no arm64 support`
- Add in latest pyarrow module: `pyarrow==10.0.0`

In [requirements_bundles.txt](https://github.com/getredash/redash/blob/master/requirements_bundles.txt) 

- Don't use latest v5 importlib-metadata as functions are now deprecated: `importlib-metadata==4.13.0`

I also updated the docker file to check architecture before removing db2 connectivity. You can see my repo here:
[https://github.com/willgladstone/redash/tree/arm64](https://github.com/willgladstone/redash/tree/arm64)

**Comment by arikfr** [comment]
Thanks, @willgladstone . I used your suggestions in #5932.
